misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractCharSequenceToContainNotToContainExpectationsTest.kt

```diff
@@ -0,0 +1,223 @@
+package ch.tutteli.atrium.specs.integration
+
+import ch.tutteli.atrium.api.fluent.en_GB.*
+import ch.tutteli.atrium.api.verbs.internal.expect
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.logic._logic
+import ch.tutteli.atrium.specs.*
+import ch.tutteli.atrium.testfactories.TestFactory
+import ch.tutteli.atrium.translations.DescriptionCharSequenceExpectation
+import ch.tutteli.atrium.specs.integration.CharSequenceToContainSpecBase.Companion.text
+import ch.tutteli.atrium.specs.integration.CharSequenceToContainSpecBase.Companion.noMatchFoundDescr
+
+@Suppress("FunctionName")
+abstract class AbstractCharSequenceToContainNotToContainExpectationsTest(
+    private val toContainSpec: Fun2<CharSequence, Any, Array<out Any>>,
+    private val notToContainSpec: Fun2<CharSequence, Any, Array<out Any>>,
+) : ExpectationFunctionBaseTest() {
+
+    @TestFactory
+    fun subjectLessTest() = subjectLessTestFactory(
+        toContainSpec.forSubjectLessTest("hello", arrayOf()),
+        notToContainSpec.forSubjectLessTest("hello", arrayOf())
+    )
+
+    fun Expect<CharSequence>.toContainFun(t: String, vararg tX: String) = toContainSpec.invoke(this, t, tX)
+
+    fun Expect<CharSequence>.notToContainFun(t: String, vararg tX: String) = notToContainSpec.invoke(this, t, tX)
+
+    val valueWithIndent = "$indentRootBulletPoint$listBulletPoint$value"
+
+    @TestFactory
+    fun toContain__subject_empty_string() = testFactory(toContainSpec) { toContainFun ->
+        val fluentEmptyString = expect("" as CharSequence)
+
+        it("${toContainSpec.name} 'Hello' throws AssertionError") {
+            expect {
+                fluentEmptyString.toContainFun("Hello")
+            }.toThrow<AssertionError> {
+                messageToContain(
+                    "$rootBulletPoint$toContainDescr: $separator" +
+                        "$valueWithIndent: \"Hello\"",
+                    noMatchFoundDescr
+                )
+            }
+        }
+    }
+
+    @TestFactory
+    fun notToContain__subject_empty_string() = testFactory(notToContainSpec) { notToContainFun ->
+        val fluentEmptyString = expect("" as CharSequence)
+
+        it("${notToContainSpec.name} 'Hello' does not throw") {
+            fluentEmptyString.notToContainFun("Hello")
+        }
+    }
+
+    @TestFactory
+    fun toContain__subject_text_search_for_hello_and_robert() = testFactory(toContainSpec) { toContainFun ->
```

Comment:
not sure, would it make sense to rename to the following to make it clear in which case we search? (of course then you would need to put `'hello' and 'robert' into an own method
```suggestion
    fun toContain__subject_text_search_for_Hello_and_Robert() = testFactory(toContainSpec) { toContainFun ->
```

Reply:
Yea, I think so. In this sense let me split Hello and notInThere as well.